pygmt/src/grdcut.py

```diff
@@ -133,9 +154,27 @@ def grdcut(
     with Session() as lib:
         with (
             lib.virtualfile_in(check_kind="raster", data=grid) as vingrd,
+            lib.virtualfile_in(
+                check_kind="vector", data=polygon, required=False
+            ) as vinpoly,
             lib.virtualfile_out(kind=outkind, fname=outgrid) as voutgrd,
         ):
             aliasdict["G"] = voutgrd
+
+            if polygon is not None:
+                if isinstance(polygon, PathLike) or hasattr(
+                    polygon, "__geo_interface__"
+                ):
+                    aliasdict["F"] = str(vinpoly) + aliasdict.get("F", "")
+                else:
+                    polygon_type = type(polygon).__name__
+                    err_msg = "Invalid polygon type"
+                    raise GMTValueError(
+                        err_msg,
+                        polygon_type,
+                        "Must be a PathLike, GeoDataFrame, or Shapely geometry.",
+                    )
```

Comment:
Please see my comment above. The check can be done earlier.
```suggestion
            if polygon:
                aliasdict["F"] = str(vinpoly) + aliasdict.get("F", "")
```